### ISSUE TITLE
Possible minor bug: calling scheduler.step() on test_dataloader

### ISSUE THREAD

**Issue by Luvata** [NVlabs/PALAVRA#4]
After a quick look, I guess there should be a check for the flag `is_train` before calling `scheduler.step()` in [line 255](https://github.com/NVlabs/PALAVRA/blob/main/get_f_theta.py#L255), so that we won't accidentally step the scheduler when evaluating on test set. What do you think @yuvalatzmon ?

**Comment by NivC** [comment]
Dear Luvata,

Thank you for the interest in out work and apologies for the time it took us to reply.

Good catch!

This may indeed change the effective hyperparameters values of the used by the scheduler. This is equivalent to choosing a slightly different `step_size`.

Best,
Niv.
